src/app/locator/qgsnominatimlocatorfilter.cpp

```diff
@@ -50,3 +52,30 @@ void QgsNominatimLocatorFilter::triggerResult( const QgsLocatorResult &result )
   }
   QgsGeocoderLocatorFilter::triggerResult( result );
 }
+
+void QgsNominatimLocatorFilter::openConfigWidget( QWidget *parent )
+{
+  auto dlg = std::make_unique<QDialog>( parent );
+  dlg->setWindowTitle( "Nominatim Geocoder Country Codes" );
+
+  QFormLayout *formLayout = new QFormLayout;
+  QLineEdit *countryCodesEdit = new QLineEdit( dlg.get() );
+
+  // Load existing settings
+  QgsSettings settings;
+  countryCodesEdit->setText( settings.value( "locator_filters/nominatim_geocoder/country_codes", "", QgsSettings::App ).toString() );
```

Comment:
Maybe use new settings tree API here and in other places?

Reply:
Don't know, I used the code from other Locator filters as example. Like this https://github.com/qgis/QGIS/blob/f38d190f56f691ace1ddb779a4597c6aefb0ec7f/src/app/locator/qgsalllayersfeatureslocatorfilter.cpp#L219
I can check this, but not sure what will be better. Is approach I used deprecated?

Reply:
It is not deprecated, but IMHO for new code we should use new API. Also with this API new setting will be nicely exposed in the advanced settings editor.

Reply:
I see what you mean now, yes it makes sense for new code to use new setting. I will look at that, but first trying to fix override position of fetchResults below...

Reply:
I now see also documentation to use [settingsTree](https://docs.qgis.org/3.40/en/docs/developers_guide/codingstandards.html#settings), so thanks for pointing out and I should have check this before.